packages/salesforce-adapter/src/metadata_deploy.ts

```diff
@@ -385,6 +399,41 @@ const quickDeployOrDeploy = async (
 const isQuickDeployable = (deployRes: SFDeployResult): boolean =>
   deployRes.id !== undefined && deployRes.checkOnly && deployRes.success && deployRes.numberTestsCompleted >= 1
 
+const isEmptyValue = (value: Value): boolean => {
+  if (_.isArray(value)) {
+    return value.length === 0
+  }
+  if (_.isPlainObject(value)) {
+    return Object.keys(value).length === 0
+  }
+  return value === undefined
+}
+
+const mapNestedNamesToElemIds = (nestedType: TypeElement, nestedNames: string[]): NameToElemIDMap => (
+  Object.fromEntries(
+    nestedNames
+      .map(nestedName => [
+        nestedName,
+        nestedType.elemID.createNestedID('instance', naclCase(nestedName)),
+      ])
+  )
+)
+
+const getExistingNestedFields = (
+  instance: MetadataInstanceElement,
+  nestedTypeInfo: NestedMetadataTypeInfo
+): {
+  nestedType: TypeElement
+  nestedNames: string[]
+}[] => (
+  nestedTypeInfo.nestedInstanceFields
+    .filter(field => !isEmptyValue(instance.value[field]))
+    .map(field => ({
+      nestedType: getTypeOfNestedElement(instance, field),
+      nestedNames: getNamesOfNestedElements(instance, field),
+    }))
```

Comment:
The method `isEmptyValue` is redundant here no?

```suggestion
    .map(field => 
      nestedNames = getNamesOfNestedElements(instance, field)
      return nestedNames.length === 0
      ? undefined
      :  {
        nestedType: getTypeOfNestedElement(instance, field),
        nestedNames: getNamesOfNestedElements(instance, field),
    }).filter(isDefined)
```

Reply:
Ack, though I ended up implementing it a bit differently.